scripts/machine_translation/train_transformer.py

```diff
@@ -348,35 +487,56 @@ def train(args):
                                                     alpha=args.label_smooth_alpha,
                                                     from_logits=False)
     label_smooth_loss.hybridize()
-    rescale_loss = 100.0
-    
+
+    # Construct the beam search sampler
+    scorer = BeamSearchScorer(alpha=args.lp_alpha,
+                              K=args.lp_k,
+                              from_logits=False)
+    beam_search_sampler = BeamSearchSampler(beam_size=args.beam_size,
```

Comment:
I wonder if this should be done outside of `def train`. Current training is done through MLE so decoding doesn't seem needed.

Reply:
It's currently included because we will need to report the validation BLEU score.

Reply:
should it be part of evaluation method?

Reply:
Since validation is part of the training process, it should be better to be created in the train function.